lib/apiservers/engine/backends/container.go

```diff
@@ -219,53 +219,74 @@ func (c *Container) TaskInspect(cid, cname, eid string) (*models.TaskInspectResp
 
 }
 
+func (c *Container) TaskWait(cid, cname, eid string) error {
+	// obtain a portlayer client
+	client := c.containerProxy.Client()
+
+	handle, err := c.Handle(cid, cname)
+	if err != nil {
+		return err
+	}
+
+	// wait the Task to start
+	config := &models.TaskWaitConfig{
+		Handle: handle,
+		ID:     eid,
+	}
+
+	params := tasks.NewWaitParamsWithContext(ctx).WithConfig(config)
```

Comment:
I think this is my misconception about the meaning of task.Wait - at this point I was thinking it was waiting for task exit.